pkg/helm/handlers/handler_test.go

```diff
@@ -290,7 +330,202 @@ func TestHelmHandlers_HandleGetChart(t *testing.T) {
+	tests := []struct {
+		name                string
+		expectedResponse    string
+		uninstallResponse   *release.UninstallReleaseResponse
+		expectedContentType string
+		error
+		httpStatusCode int
+	}{
+		{
+			name:                "Invalid chart uninstall release test",
+			error:               errors.New("Chart path is invalid"),
+			expectedResponse:    `{"error":"Failed to uninstall helm release: Chart path is invalid"}`,
+			httpStatusCode:      http.StatusBadGateway,
+			expectedContentType: "application/json",
+		},
+		{
+			name:                "Valid chart uninstall release test",
+			expectedResponse:    `{"release":{"name":"Test"}}`,
+			uninstallResponse:   fakeUninstallResponse,
+			httpStatusCode:      http.StatusOK,
+			expectedContentType: "application/json",
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			handlers := fakeHelmHandler()
+			handlers.uninstallRelease = fakeUninstallRelease(tt.uninstallResponse, tt.error)
+
+			request := httptest.NewRequest("", "/foo", strings.NewReader("{}"))
+			response := httptest.NewRecorder()
+
+			handlers.HandleUninstallRelease(&auth.User{}, response, request)
+
+			if response.Code != tt.httpStatusCode {
+				t.Errorf("response code should be %v but got %v", tt.httpStatusCode, response.Code)
+			}
+			if response.Header().Get("Content-Type") != tt.expectedContentType {
+				t.Errorf("content type should be %s but got %s", tt.expectedContentType, response.Header().Get("Content-Type"))
+			}
+			if response.Body.String() != tt.expectedResponse {
+				t.Errorf("response body not matching expected is %s and received is %s", tt.expectedResponse, response.Body.String())
+			}
+		})
+	}
+}
+
+func TestHelmHandlers_HandleHelmRollbackRelease(t *testing.T) {
+	tests := []struct {
+		name                string
+		expectedResponse    string
+		fakeResp            map[string]interface{}
+		expectedContentType string
+		error
+		httpStatusCode int
+	}{
+		{
+			name:                "Invalid chart rollback release test",
+			error:               errors.New("Chart path is invalid"),
+			expectedResponse:    `{"error":"Failed to rollback helm releases: Chart path is invalid"}`,
+			httpStatusCode:      http.StatusBadGateway,
+			expectedContentType: "application/json",
+		},
+		{
+			name:             "Valid chart rollback release test",
+			expectedResponse: `{"status":"rollback successful"}`,
+			fakeResp: map[string]interface{}{
+				"status": "rollback successful",
+			},
+			expectedContentType: "application/json",
+			error:               nil,
+			httpStatusCode:      http.StatusOK,
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			handlers := fakeHelmHandler()
+			var request *http.Request
+
+			handlers.rollbackRelease = fakeRollbackRelease(tt.fakeResp, tt.error)
+
+			request = httptest.NewRequest("", "/foo?revision=1", strings.NewReader(`{"name": "test", "namespace":"test"}`))
```

Comment:
the payload should provided in body, as json, not as query param.

Reply:
we should also test what is going on if the payload has wrong data, or no version specified.

Reply:
Done